iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalCatalog.java

```diff
@@ -57,6 +63,18 @@ protected TableOperations newTableOps(TableIdentifier tableIdentifier) {
         new MetricsReporter(this.meterRegistry, METRICS_PREFIX, Lists.newArrayList()));
   }
 
+  /** Overwritten for annotation purpose. */
+  @Override
+  @IcebergFeatureGate(value = STOP_TABLE_CREATION_TOGGLE)
+  public Table createTable(
+      TableIdentifier identifier,
+      Schema schema,
+      PartitionSpec spec,
+      String location,
+      Map<String, String> properties) {
+    return super.createTable(identifier, schema, spec, location, properties);
```

Comment:
I don't see an easy way to use the annotation feature on Storage Selectors. We have Selector interface as follows:

````
Class StorageSelector {
    Storage selectStorage(db, table) {
            isFeatureEnabled = ...      // planning to call FeatureToggle
            if (isFeatureEnabled) {
                    return Storage;
            }
           return storageManager.defaultstorage()
    }
}
````

We can always use `@ToggleStatusesRepository` but not the annotation easily. Let me know if there's some way

Reply:
See this: https://github.com/linkedin/openhouse/pull/174#discussion_r1747542004 
But yes annotation is just a syntactic sugar.